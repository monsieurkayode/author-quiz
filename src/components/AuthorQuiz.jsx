import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { getTurnDataAction } from '../actions/quizActions';
import Hero from './Hero';
import Turn from './Turn';
import Continue from './Continue';
import Footer from './Footer';

import '../styles/AuthorQuiz.css';

class AuthorQuiz extends Component {
  componentWillMount() {
    this.props.getTurnDataAction();
  }

  render() {
    return (
      <div className="container-fluid">
        <Hero />
        <Turn />
        <Continue />
        <p><Link to="/add" >Add an author</Link></p>
        <Footer />
      </div>
    );
  }
}

export default connect(null, { getTurnDataAction })(AuthorQuiz);
