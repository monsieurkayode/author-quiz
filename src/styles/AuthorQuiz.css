@import 'bootstrap.min.css';

.header {
  padding: 2rem;
}

.turn > div {
  height: 400px;
}

.turn {
  padding: 20px 0;
}

.authorimage { 
  max-height: 100%;
  width: 100%;
  max-width: 260px;
}

.answer {
  background-color: #F0F7FD;
  border-color: #D0E3F0;
  margin-bottom: 20px;
  padding: 15px 30px 15px;
  border-left: 5px solid #D0E3F0;
  cursor: pointer;
  transition: background-color .3s linear;
}

.answer:hover {
  background-color: #D0E3F0;
}

.continue {
  padding-top: 20px;
}
