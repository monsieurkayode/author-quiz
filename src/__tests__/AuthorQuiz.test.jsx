import React from 'react';
import { render } from 'react-dom';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AuthorQuiz from '../components/AuthorQuiz';

configure({ adapter: new Adapter() });

const onAnswerSelected = jest.fn();

let state = {
  turnData: {
    author: {
      name: 'Mark Twain',
      imageUrl: 'images/authors/marktwain.jpg',
      imageSource: 'Wikimedia Commons',
      books: [
        'The Adventures of Huckleberry Finn',
        'Life Of Mississippi',
        'Roughing It'
      ]
    },
    books: [
      'The Adventures of Huckleberry Finn',
      'Life Of Mississippi',
      'Roughing It'
    ],
  },
  highlight: 'none',
}

describe('Author Quiz Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    render(<AuthorQuiz {...state} onAnswerSelected={onAnswerSelected} />, div);
  });

  it('renders correctly', () => {
    const wrapper = shallow(<AuthorQuiz
      {...state}
      onAnswerSelected={onAnswerSelected}
    />)
    expect(wrapper).toMatchSnapshot();
  });

  describe('When no answer has been selected', () => {
    let wrapper;
    beforeAll(() => {
      wrapper = mount(<AuthorQuiz
        {...state}
        onAnswerSelected={onAnswerSelected}
      />);
    });

    it('should have no background color', () => {
      expect(wrapper.find('div.row.turn').props().style.background)
        .toBe('');
    });
  });

  describe('When the wrong answer has been selected', () => {
    let wrapper;
    beforeAll(() => {
      state = {...state, highlight: 'wrong'}
      wrapper = mount(<AuthorQuiz
        {...state}
        onAnswerSelected={onAnswerSelected}
      />);
    });
    
    it('should have a "#F55" background color', () => {
      expect(wrapper.find('div.row.turn').props().style.background)
        .toBe('#F55')
    });
  });

  describe('When the right amswer has been selected', () => {
    let wrapper;
    beforeAll(() => {
      state = {...state, highlight: 'right'};
      wrapper = mount(<AuthorQuiz 
        {...state}
        onAnswerSelected={onAnswerSelected}
      />);
    });

    it('should have a "#3FA" background color', () => {
      expect(wrapper.find('div.row.turn').props().style.background)
        .toBe('#3FA');
    })
  });

  describe('When the first answer is selected', () => {
    let wrapper;
    beforeAll(() => {
      wrapper = mount(<AuthorQuiz 
        {...state}
        onAnswerSelected={onAnswerSelected}
      />);
      wrapper.find('.answer').first().simulate('click');
    });

    it('onAnswerSelected should be called', () => {
      expect(onAnswerSelected).toHaveBeenCalled();
    });

    it('should receive The Shining', () => {
      expect(onAnswerSelected)
        .toHaveBeenCalledWith('The Adventures of Huckleberry Finn');
    })
  });
});