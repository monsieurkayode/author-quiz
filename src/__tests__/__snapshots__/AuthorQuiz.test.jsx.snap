// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author Quiz Component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AuthorQuiz
    highlight="none"
    onAnswerSelected={[Function]}
    turnData={
        Object {
            "author": Object {
              "books": Array [
                "The Adventures of Huckleberry Finn",
                "Life Of Mississippi",
                "Roughing It",
              ],
              "imageSource": "Wikimedia Commons",
              "imageUrl": "images/authors/marktwain.jpg",
              "name": "Mark Twain",
            },
            "books": Array [
              "The Adventures of Huckleberry Finn",
              "Life Of Mississippi",
              "Roughing It",
            ],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Hero />,
        <Turn
          author={
                    Object {
                              "books": Array [
                                "The Adventures of Huckleberry Finn",
                                "Life Of Mississippi",
                                "Roughing It",
                              ],
                              "imageSource": "Wikimedia Commons",
                              "imageUrl": "images/authors/marktwain.jpg",
                              "name": "Mark Twain",
                            }
          }
          books={
                    Array [
                              "The Adventures of Huckleberry Finn",
                              "Life Of Mississippi",
                              "Roughing It",
                            ]
          }
          highlight="none"
          onAnswerSelected={[Function]}
/>,
        <Continue />,
        <Footer />,
      ],
      "className": "container-fluid",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "author": Object {
            "books": Array [
              "The Adventures of Huckleberry Finn",
              "Life Of Mississippi",
              "Roughing It",
            ],
            "imageSource": "Wikimedia Commons",
            "imageUrl": "images/authors/marktwain.jpg",
            "name": "Mark Twain",
          },
          "books": Array [
            "The Adventures of Huckleberry Finn",
            "Life Of Mississippi",
            "Roughing It",
          ],
          "highlight": "none",
          "onAnswerSelected": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Hero />,
          <Turn
            author={
                        Object {
                                    "books": Array [
                                      "The Adventures of Huckleberry Finn",
                                      "Life Of Mississippi",
                                      "Roughing It",
                                    ],
                                    "imageSource": "Wikimedia Commons",
                                    "imageUrl": "images/authors/marktwain.jpg",
                                    "name": "Mark Twain",
                                  }
            }
            books={
                        Array [
                                    "The Adventures of Huckleberry Finn",
                                    "Life Of Mississippi",
                                    "Roughing It",
                                  ]
            }
            highlight="none"
            onAnswerSelected={[Function]}
/>,
          <Continue />,
          <Footer />,
        ],
        "className": "container-fluid",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "author": Object {
              "books": Array [
                "The Adventures of Huckleberry Finn",
                "Life Of Mississippi",
                "Roughing It",
              ],
              "imageSource": "Wikimedia Commons",
              "imageUrl": "images/authors/marktwain.jpg",
              "name": "Mark Twain",
            },
            "books": Array [
              "The Adventures of Huckleberry Finn",
              "Life Of Mississippi",
              "Roughing It",
            ],
            "highlight": "none",
            "onAnswerSelected": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
